pkg/hooks/manager.go

```diff
@@ -187,51 +187,59 @@ func (m *Manager) OnDefineDomain(domainSpec *virtwrapApi.DomainSpec, vmi *v1.Vir
 	if err != nil {
 		return "", fmt.Errorf("Failed to marshal domain spec: %v", domainSpec)
 	}
-	if callbacks, found := m.CallbacksPerHookPoint[hooksInfo.OnDefineDomainHookPointName]; found {
-		for _, callback := range callbacks {
-			if callback.Version == hooksV1alpha1.Version || callback.Version == hooksV1alpha2.Version {
-				vmiJSON, err := json.Marshal(vmi)
-				if err != nil {
-					return "", fmt.Errorf("Failed to marshal VMI spec: %v", vmi)
-				}
 
-				conn, err := grpcutil.DialSocketWithTimeout(callback.SocketPath, 1)
-				if err != nil {
-					log.Log.Reason(err).Infof(dialSockErr, callback.SocketPath)
-					return "", err
-				}
-				defer conn.Close()
+	callbacks, found := m.CallbacksPerHookPoint[hooksInfo.OnDefineDomainHookPointName]
+	if !found {
+		return string(domainSpecXML), nil
+	}
 
-				ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
-				defer cancel()
+	if len(callbacks) == 0 {
+		return string(domainSpecXML), nil
+	}
 
-				switch callback.Version {
-				case hooksV1alpha1.Version:
-					client := hooksV1alpha1.NewCallbacksClient(conn)
-					result, err := client.OnDefineDomain(ctx, &hooksV1alpha1.OnDefineDomainParams{
-						DomainXML: domainSpecXML,
-						Vmi:       vmiJSON,
-					})
-					if err != nil {
-						return "", err
-					}
-					domainSpecXML = result.GetDomainXML()
-				case hooksV1alpha2.Version:
-					client := hooksV1alpha2.NewCallbacksClient(conn)
-					result, err := client.OnDefineDomain(ctx, &hooksV1alpha2.OnDefineDomainParams{
-						DomainXML: domainSpecXML,
-						Vmi:       vmiJSON,
-					})
-					if err != nil {
-						return "", err
-					}
-					domainSpecXML = result.GetDomainXML()
-				default:
-					panic("Should never happen, version compatibility check is done during Info call")
-				}
-			}
+	callback := callbacks[0]
```

Comment:
This will handle only the first callback while originally there was a loop over the whole array. That does not seem correct. Or am I missing smth?

Reply:
Ups, you are correct, we should call `client.OnDefineDomain` for all the specs.